### ISSUE TITLE
[Spec] metadata missing from classDeclaration

### ISSUE THREAD

**Issue by kaby76** [dart-lang/language#2279]
From the [spec](https://github.com/dart-lang/language/blob/8fb9f0ad4e8e5eabb1ec83abcd53f2f587d111fb/specification/dartLangSpec.tex#L2404), a `classDeclaration` is:
```
classDeclaration : 'abstract'? 'class' typeIdentifier typeParameters? superclass? interfaces? '{' ( metadata classMemberDeclaration )* '}' | 'abstract'? 'class' mixinApplicationClass ;
```
Unfortunately, `metadata` is missing (`classDeclaration : metadata 'abstract'? 'class' typeIdentifier typeParameters? superclass? interfaces? '{' ( metadata classMemberDeclaration )* '}' | 'abstract'? 'class' mixinApplicationClass ;`), so it cannot parse [this](https://github.com/dart-lang/sdk/blob/d028c0e3c191ee78577a44a25a9d7ea2682355b2/sdk/lib/async/deferred_load.dart#L12).

**Comment by kaby76** [comment]
Actually, `classDeclaration` shouldn't be modified as I suggested above. Looking over how [libraryDeclation](https://github.com/dart-lang/language/blob/8fb9f0ad4e8e5eabb1ec83abcd53f2f587d111fb/specification/dartLangSpec.tex#L18871) contains `(metadata topLevelDeclaration)*`, [partDeclaration](https://github.com/dart-lang/language/blob/8fb9f0ad4e8e5eabb1ec83abcd53f2f587d111fb/specification/dartLangSpec.tex#L19775) should be the same, `partDeclaration : partHeader (metadata topLevelDeclaration)* EOF ;`, so [this declaration can be parsed](https://github.com/dart-lang/sdk/blob/a89370324e4f65b99b1893963d8d74558417b2ea/sdk/lib/cli/wait_for.dart#L49). This corresponds to the [parser code](https://github.com/dart-lang/sdk/blob/a89370324e4f65b99b1893963d8d74558417b2ea/pkg/_fe_analyzer_shared/lib/src/parser/parser_impl.dart#L494). (This is why I need to scrape the grammar directly from the implementation.)

**Comment by eernstg** [comment]
Right, `<libraryDeclaration>` includes the `<metadata>`, but `<partDeclaration>` needs to be fixed, thanks!
Done in https://github.com/dart-lang/language/pull/2280.
